File: src/main/java/com/anod/appwatcher/tags/AppsTagActivity.java
package com.anod.appwatcher.tags;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.TabLayout;
import android.support.v7.graphics.Palette;
import android.view.MenuItem;

import com.anod.appwatcher.R;
import com.anod.appwatcher.model.Filters;
import com.anod.appwatcher.model.Tag;
import com.anod.appwatcher.ui.AppWatcherBaseActivity;

import java.util.ArrayList;

/**
 * @author algavris
 * @date 18/03/2017.
 */

public class AppsTagActivity extends AppWatcherBaseActivity {
    public static final String EXTRA_TAG = "extra_tag";

    private Tag mTag;

    public static Intent createTagIntent(Tag tag, Context context) {
        Intent intent = new Intent(context, AppsTagActivity.class);
        intent.putExtra(EXTRA_TAG, tag);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        mTag = getIntentExtras().getParcelable(EXTRA_TAG);
        assert mTag != null;
        super.onCreate(savedInstanceState);

        AppBarLayout appBarLayout = (AppBarLayout) findViewById(R.id.appbar);
        appBarLayout.setBackgroundColor(mTag.color);

        float[] hsv = new float[3];
        Color.colorToHSV(mTag.color, hsv);
        hsv[2] *= 0.6f;
        int dark = Color.HSVToColor(hsv);
        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabs);
        tabLayout.setSelectedTabIndicatorColor(dark);

        setTitle(mTag.name);
    }

    @Override
    protected int getContentLayout() {
        return R.layout.activity_main;
    }

    @Override
    protected int getMenuResource() {
        return R.menu.main_tag;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.menu_act_addtag)
        {
            startActivity(AppsTagSelectActivity.createIntent(mTag, this));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    protected AppWatcherBaseActivity.Adapter createViewPagerAdapter() {
        AppWatcherBaseActivity.Adapter adapter = new AppWatcherBaseActivity.Adapter(getSupportFragmentManager());
        adapter.addFragment(AppsTagListFragment.newInstance(
                Filters.TAB_ALL,
                mPreferences.getSortIndex(),
                new AppsTagListFragment.DefaultSection(),
                mTag), getString(R.string.tab_all));
        adapter.addFragment(AppsTagListFragment.newInstance(
                Filters.TAB_INSTALLED,
                mPreferences.getSortIndex(),
                new AppsTagListFragment.DefaultSection(),
                mTag), getString(R.string.tab_installed));
        adapter.addFragment(AppsTagListFragment.newInstance(
                Filters.TAB_UNINSTALLED,
                mPreferences.getSortIndex(),
                new AppsTagListFragment.DefaultSection(),
                mTag), getString(R.string.tab_not_installed));
        return adapter;
    }

    @Override
    protected boolean isDrawerEnabled() {
        return false;
    }
}

File: app/src/main/java/com/anod/appwatcher/tags/AppsTagActivity.kt
package com.anod.appwatcher.tags

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import com.anod.appwatcher.R
import com.anod.appwatcher.database.entities.Tag
import com.anod.appwatcher.model.Filters
import com.anod.appwatcher.utils.prefs
import com.anod.appwatcher.watchlist.WatchListActivity
import com.google.android.material.appbar.MaterialToolbar
import info.anodsplace.framework.app.CustomThemeColors
import info.anodsplace.framework.app.addMultiWindowFlags
import org.koin.core.component.KoinComponent

/**
 * @author Alex Gavrishev
 * *
 * @date 18/03/2017.
 */

class AppsTagActivity : WatchListActivity(), KoinComponent {
    private var tag: Tag = Tag("")

    override val menuResource: Int
        get() = R.menu.tagslist

    override val themeRes: Int
        get() = if (themeColors.statusBarColor.isLight)
            theme.themeLightActionBar
        else
            theme.themeDarkActionBar

    override val themeColors: CustomThemeColors
        get() = CustomThemeColors(tag.color, theme.colors.navigationBarColor)

    override fun onCreate(savedInstanceState: Bundle?) {
        tag = restoreTag(savedInstanceState)
        super.onCreate(savedInstanceState)

        val toolbar = binding.toolbar as MaterialToolbar
        toolbar.setBackgroundColor(tag.color)
        title = tag.name
    }

    private fun restoreTag(savedInstanceState: Bundle?): Tag {
        if (intentExtras.containsKey(EXTRA_TAG)) {
            return intentExtras.getParcelable(EXTRA_TAG) ?: Tag("")
        } else if (savedInstanceState != null) {
            val savedTag = savedInstanceState.getParcelable<Tag?>(EXTRA_TAG)
            if (savedTag != null) {
                return savedTag
            }
        }
        return Tag("")
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putParcelable(EXTRA_TAG, tag)
        super.onSaveInstanceState(outState)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.menu_act_addtag) {
            AppsTagSelectDialog.show(tag, supportFragmentManager)
            return true
        }
        if (item.itemId == R.id.menu_act_edittag) {
            EditTagDialog.show(supportFragmentManager, tag, theme)
        }
        return super.onOptionsItemSelected(item)
    }

    override fun createViewPagerAdapter(): Adapter {
        val factories = listOf(
                AppsTagListFragment.Factory(Filters.TAB_ALL, prefs.sortIndex, tag, title = getString(R.string.tab_all)),
                AppsTagListFragment.Factory(Filters.INSTALLED, prefs.sortIndex, tag, title = getString(R.string.tab_installed)),
                AppsTagListFragment.Factory(Filters.UNINSTALLED, prefs.sortIndex, tag, title = getString(R.string.tab_not_installed)),
                AppsTagListFragment.Factory(Filters.UPDATABLE, prefs.sortIndex, tag, title = getString(R.string.tab_updatable))
        )
        return Adapter(factories, this)
    }

    companion object {
        const val EXTRA_TAG = "extra_tag"

        fun createTagIntent(tag: Tag, context: Context) = Intent(context, AppsTagActivity::class.java).apply {
            putExtra(EXTRA_TAG, tag)
            addMultiWindowFlags(context)
        }
    }
}